### PR TITLE
Add add-homepage-1 to master

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently, the ScrollToTop fix is not applied.
- Currently, the page consists of the template.
- Currently, there are no custom sections.
- Currently, there is no development warning page.

**What is the new behavior (if this is a feature change)?**
- The ScrollToTop fix has been applied.
- The page has been rebuilt by design and functionality of the [Prototype](https://github.com/aichner/Company-Homepage-Prototype).
- All the sections required for the final homepage have been added.
- A development warning has been designed. It is only visible, if not on localhost or if the website is in development mode.
- The 2020 coding style has been enforced.

**Other information**:
- This is version 1.0.0 of the Company homepage. It has already been deployed.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,26 +9,43 @@ import { BrowserRouter as Router } from "react-router-dom";
...
-            <Routes />
-          </main>
-          <Footer />
-        </div>
-      </Router>
-    );
+    if (
+      window.location.hostname === "localhost" ||
+      window.location.hostname === "127.0.0.1" ||
+      isLive
```
Comment (pinterid) [inline src/App.js:31]: isLive could be changes to the real company domain.

File: src/components/molecules/Navbar/index.jsx [hunk 2]
```diff
@@ -33,6 +44,17 @@ class Navbar extends React.Component {
     this.state.collapseID === collapseID && this.setState({ collapseID: "" });
   };
 
+  // Get navbar mode
+  _getMode = () => {
```
Comment (Aichnerc) [inline src/components/molecules/Navbar/index.jsx]: Please do not start functions with `_`.
Change to: `getMode = () => {`

File: src/components/molecules/Footer/footer.scss [hunk 3]
```diff
@@ -1,25 +1,136 @@
...
+
+  a {
+    color: #6c757d !important;
     &:hover {
-      background: rgba(0, 0, 0, 0.2);
+      color: lighten(#6c757d, 10%) !important;
+    }
+  }
+
+  // List item styleF
```
Comment (pinterid) [inline src/components/molecules/Footer/footer.scss]: styleF ??

File: src/components/molecules/Footer/footer.scss [hunk 4]
```diff
@@ -1,25 +1,136 @@
...
+      &#logo {
+        .st0 {
+          fill: #ffffff;
+        }
+        .st1 {
+          fill: #f61a42;
+        }
+      }
+    }
+  }
```
Comment (pinterid) [inline src/components/molecules/Footer/footer.scss:110]: Missing new line.

File: src/components/molecules/Footer/index.jsx [hunk 5]
```diff
@@ -12,96 +12,306 @@ import {
...
+              rel="noopener noreferrer"
+              target="_blank"
+            >
+              <MDBIcon fab icon="twitter" />
+            </MDBBtn>
+            <MDBBtn
+              tag="a"
+              floating
+              social="wa"
+              href="https://api.whatsapp.com/send?phone=4368120502754&text=Hey%2C%20ich%20bin%20interessiert%20an%20einem%20Angebot%20eurer%20Agentur."
```
Comment (pinterid) [inline src/components/molecules/Footer/index.jsx]: To many chars in one row, consider using multiline string.

File: src/components/molecules/Navbar/index.jsx [hunk 6]
```diff
@@ -13,15 +14,25 @@ import {
...
+
+//> CSS
+import "./navbar.scss";
 
 class Navbar extends React.Component {
-  state = {
-    collapseID: "",
-  };
+  constructor(props) {
+    super(props);
```
Comment (pinterid) [inline src/components/molecules/Navbar/index.jsx:31]: Missing new line.

File: src/components/molecules/Navbar/index.jsx [hunk 7]
```diff
@@ -33,6 +44,17 @@ class Navbar extends React.Component {
     this.state.collapseID === collapseID && this.setState({ collapseID: "" });
   };
 
+  // Get navbar mode
+  _getMode = () => {
+    let opts = {};
```
Comment (pinterid) [inline src/components/molecules/Navbar/index.jsx:50]: Missing new line.

File: src/components/molecules/Navbar/index.jsx [hunk 8]
```diff
@@ -33,6 +44,17 @@ class Navbar extends React.Component {
...
+    if (this.props.darkMode) {
+      opts["dark"] = "dark";
+    } else {
+      opts["light"] = "light";
+    }
```
Comment (pinterid) [inline src/components/molecules/Navbar/index.jsx:56]: Missing new line.

File: src/components/molecules/Navbar/index.jsx [hunk 9]
```diff
@@ -42,14 +64,25 @@ class Navbar extends React.Component {
       />
     );
 
+    // Get react-router-dom location and navbar collapseID
+    const { location } = this.props;
     const { collapseID } = this.state;
+
+    // Debugging
+    console.log(location);
```
Comment (pinterid) [inline src/components/molecules/Navbar/index.jsx]: Is this console.log still required?

File: src/components/molecules/Navbar/index.jsx [hunk 10]
```diff
@@ -88,9 +140,9 @@ class Navbar extends React.Component {
   }
 }
 
-export default Navbar;
+export default withRouter(Navbar);
 
 /**
  * SPDX-License-Identifier: (EUPL-1.2)
- * Copyright © 2019-2020 Werbeagentur Christian Aichner
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline src/components/molecules/Navbar/index.jsx]: Missing -2020.

File: src/components/molecules/Navbar/navbar.scss [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+          fill: $agencyRed;
+        }
+      }
+    }
+  }
+}
+
+/** 
+* SPDX-License-Identifier: (EUPL-1.2)
+* Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline src/components/molecules/Navbar/navbar.scss]: Missing -2020

File: src/components/molecules/Navbar/index.jsx [hunk 12]
```diff
@@ -42,14 +64,25 @@ class Navbar extends React.Component {
...
+
+    // Debugging
+    console.log(location);
+
     return (
       <div>
-        <MDBNavbar color="white" light expand="md" fixed="top" scrolling>
+        <MDBNavbar
+          color={this.props.darkMode ? "agency-dark" : "white"}
+          {...this._getMode()}
```
Comment (Aichnerc) [inline src/components/molecules/Navbar/index.jsx]: Alter to `...this.getMode()`

File: src/components/organisms/sections/CallToAction/index.jsx [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+                </MDBView>
+              </MDBCol>
+              <MDBCol></MDBCol>
+            </MDBRow>
+          </MDBCol>
+          <MDBCol md="auto" className="text-center">
+            <h2 className="gidole">
+              Es ist einfach, <strong>jetzt</strong> zu starten!
+            </h2>
+            <p>Und der Einstieg und Erstberatung sind kostenlos.</p>
```
Comment (pinterid) [inline src/components/organisms/sections/CallToAction/index.jsx]: und die Ersberatung?

File: src/components/organisms/sections/CallToAction/callToAction.scss [hunk 14]
```diff
@@ -0,0 +1,113 @@
+#calltoaction {
+  overflow: hidden;
```
Comment (Aichnerc) [inline src/components/organisms/sections/CallToAction/callToAction.scss:3]: Why is this required?

File: src/components/organisms/sections/ContactForm/contactForm.scss [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      top: 50% !important;
+      left: 100% !important;
+      width: 5000px !important;
+    }
+  }
+}
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (pinterid) [inline src/components/organisms/sections/ContactForm/contactForm.scss]: Missing -2020.

File: src/components/organisms/sections/ContactForm/index.jsx [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+
+class ContactForm extends React.Component {
+  state = {
+    fullname: "",
+    email: "",
+    phone: "",
+    note: "",
+  };
+
+  componentDidMount = () => {};
```
Comment (pinterid) [inline src/components/organisms/sections/ContactForm/index.jsx]: Is this required?

File: src/components/organisms/sections/CallToAction/index.jsx [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+import { Parallax } from "react-scroll-parallax";
+
+//> MDB
+// "Material Design for Bootstrap" is a great UI design framework
+import { MDBRow, MDBCol, MDBBtn, MDBView, MDBIcon, MDBMask } from "mdbreact";
+
+//> Images
+// Rocket
+import rocket from "../../../../assets/content/icons/rocket.png";
+//import rocket_pen from '../../../../assets/content/icons/rocket_pen.png';
```
Comment (Aichnerc) [inline src/components/organisms/sections/CallToAction/index.jsx]: The `rocket_pen` is no longer required. Please remove it.

File: src/components/organisms/sections/ContactForm/index.jsx (hunk 16 above)
Comment (Aichnerc) [inline src/components/organisms/sections/ContactForm/index.jsx]: Empty function. Please remove.

File: src/components/organisms/sections/ContactForm/index.jsx [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+    return (
+      <section id="contactForm" className={darkMode ? "dark" : "light"}>
+        <MDBContainer className="py-5">
+          <MDBRow className="flex-center">
+            <MDBCol md="6">
+              <MDBCard>
+                <MDBCardBody className="py-5">
+                  <h3 className="font-weight-bold">Kontakt aufnehmen</h3>
+                  <p className="text-muted">
+                    Du hast eine Projekt-Idee und bist Dir nicht sicher um deren
```
Comment (pinterid) [inline src/components/organisms/sections/ContactForm/index.jsx]: Most of the websites write it "Projektidee" not "Projekt-Idee".

File: src/components/organisms/sections/Hero/hero.scss [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+  @media (min-width: 1200px) {
+    .container {
+      max-width: 1440px;
+    }
+  }
+}
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/organisms/sections/Hero/hero.scss]: Copyright should be 2019-2020.

File: src/components/organisms/sections/Trusted/index.jsx [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+  render() {
+    return (
+      <div id="trusted">
+        <MDBContainer className="py-5 text-center text-dark">
+          <h2 className="gidole h1-responsive font-weight-bold">
+            Viele Firmen profitieren von unseren Services
+          </h2>
+          <p className="lead">
+            Wir bieten Dir <strong>hochwertige</strong>, auf Deine Vision
+            zugeschnittene Lösungen in sämtlichen Bereichen des Marketing.
```
Comment (LucaAllmaier) [inline src/components/organisms/sections/Trusted/index.jsx]: des Marketings?

File: src/components/organisms/sections/Hero/index.jsx [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  generateRandom = () => {
+    let min = Math.ceil(10);
+    let max = Math.floor(100);
+    let arr = [1, 2, 3, 4, 5, 6, 7, 8];
+
```
Comment (Aichnerc) [inline src/components/organisms/sections/Hero/index.jsx]: Remove the space

File: src/components/organisms/sections/Services/index.jsx [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+      text: `Mit einem Imagefilm erhaltest Du ein Video, welches Du vielseitig und 
+      auf allen Plattformen einsetzen kannst.`,
+      action: {
+        text: "Erkunden",
+        color: "red",
+        link: "/services/image",
+      },
+    },
+  ],
+};
```
Comment (Aichnerc) [inline src/components/organisms/sections/Services/index.jsx]: This data could be loaded in from a seperate file.

File: src/components/organisms/sections/ContactForm/index.jsx [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+                    <MDBIcon icon="phone" />
+                    Anrufen
+                  </MDBBtn>
+                  {this.state.call && (
+                    <MDBAlert color="primary" className="mt-3">
+                      <span className="d-block">
+                        Du kannst uns täglich von 9:00 bis 18:00 telefonisch
+                        unter
+                        <br />
+                        <strong>+43 681 205 027 54</strong> erreichen.
```
Comment (pinterid) [inline src/components/organisms/sections/ContactForm/index.jsx]: Add a link on the tel number?

File: src/components/pages/BrandingPage/index.jsx [hunk 24]
```diff
@@ -0,0 +1,367 @@
...
+  MDBCard,
+  MDBCardBody,
+  MDBTable,
+  MDBTableBody,
+  MDBBtn,
+  MDBIcon,
+} from 'mdbreact';
+
+//> CSS
+import './branding.scss';
```
Comment (LucaAllmaier) [inline src/components/pages/BrandingPage/index.jsx:20]: use " "

File: src/components/organisms/sections/Trusted/index.jsx [hunk 25]
```diff
@@ -0,0 +1,103 @@
...
+      </div>
+    );
+  }
+}
+
+export default Trusted;
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/organisms/sections/Trusted/index.jsx]: Copyright should be 2019-2020.

File: src/components/organisms/sections/Trusted/trusted.scss [hunk 26]
```diff
@@ -0,0 +1,11 @@
+#trusted {
+  background: #e7e7e7;
+  img {
+    max-height: 100px;
+  }
+}
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/organisms/sections/Trusted/trusted.scss]: Copyright should be 2019-2020.

File: src/components/pages/BrandingPage/index.jsx [hunk 27]
```diff
@@ -0,0 +1,367 @@
...
+
+//> Images
+// Logo
+import { ReactComponent as Logo } from  '../../../assets/content/logo_full.svg';
```
Comment (LucaAllmaier) [inline src/components/pages/BrandingPage/index.jsx:24]: use " "

File: src/components/organisms/sections/Hero/hero.scss [hunk 28]
```diff
@@ -0,0 +1,118 @@
...
+  }
+  h6 {
+    line-height: 1.7;
+  }
+
+  // Dark mode
+  &.dark {
+    background: linear-gradient(to top, $dark 0%, darken($dark, 10%) 100%);
+    color: white;
+  }
```
Comment (pinterid) [inline src/components/organisms/sections/Hero/hero.scss:55]: Missing new line.

File: src/components/organisms/sections/Hero/hero.scss (hunk 19 above)
Comment (pinterid) [inline src/components/organisms/sections/Hero/hero.scss]: Missing -2020.

File: src/components/organisms/sections/Hero/index.jsx [hunk 29]
```diff
@@ -0,0 +1,147 @@
...
+import {
+  MDBRow,
+  MDBCol,
+  MDBBtn,
+  MDBView,
+  MDBContainer,
+  MDBIcon,
+  MDBCard,
+  MDBCardBody,
+} from "mdbreact";
```
Comment (pinterid) [inline src/components/organisms/sections/Hero/index.jsx:22]: Should be moved above MDB.

File: src/components/pages/BrandingPage/branding.scss [hunk 30]
```diff
@@ -0,0 +1,81 @@
...
+      td {
+        border-top: 1px solid rgba(0, 0, 0, 0.2);
+      }
+    }
+  }
+}
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/pages/BrandingPage/branding.scss]: Copyright should be 2019-2020.

File: src/components/pages/BrandingPage/index.jsx [hunk 31]
```diff
@@ -0,0 +1,367 @@
...
+// Wedding logo
+import weddingsLogo from '../../../assets/content/weddings.png';
+
+class BrandingPage extends React.Component {
+
```
Comment (Aichnerc) [inline src/components/pages/BrandingPage/index.jsx]: Remove the new line

File: src/components/pages/BrandingPage/index.jsx [hunk 32]
```diff
@@ -0,0 +1,367 @@
...
+  componentDidMount = () => {
+    // Set page title
+    document.title = "Evolving Our Identity";
+  }
+
+  downloadImage = (type, format) => {
+
```
Comment (Aichnerc) [inline src/components/pages/BrandingPage/index.jsx]: Remove the new line

File: src/components/pages/BrandingPage/index.jsx [hunk 33]
```diff
@@ -0,0 +1,367 @@
...
+} from 'mdbreact';
+
+//> CSS
+import './branding.scss';
+
+//> Images
+// Logo
+import { ReactComponent as Logo } from  '../../../assets/content/logo_full.svg';
+// Wedding logo
+import weddingsLogo from '../../../assets/content/weddings.png';
```
Comment (LucaAllmaier) [inline src/components/pages/BrandingPage/index.jsx]: use " "

File: src/components/organisms/sections/Hero/index.jsx (hunk 21 above)
Comment (pinterid) [inline src/components/organisms/sections/Hero/index.jsx]: Remove new line.

File: src/components/pages/BrandingPage/index.jsx [hunk 34]
```diff
@@ -0,0 +1,367 @@
...
+      </div>
+    );
+  }
+}
+
+export default BrandingPage;
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/pages/BrandingPage/index.jsx]: Copyright should be 2019-2020.

File: src/components/pages/DevelopmentPage/index.jsx [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+      </div>
+    );
+  }
+}
+
+export default DevelopmentPage;
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Christian Aichner
```
Comment (Aichnerc) [inline src/components/pages/DevelopmentPage/index.jsx]: Copyright should be 2019-2020.

File: src/components/pages/PrintingPage/index.jsx [hunk 36]
```diff
@@ -0,0 +1,258 @@
...
+      </div>
+    );
+  }
+}
+
+export default PrintingPage;
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/pages/PrintingPage/index.jsx]: Copyright should be 2019-2020.

File: src/components/pages/PrintingPage/printing.scss [hunk 37]
```diff
@@ -0,0 +1,93 @@
...
+    }
+    .stepper.timeline.stepper-vertical li:not(:last-child):after {
+      background-color: #2e2e2e;
+    }
+  }
+}
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Comment (Aichnerc) [inline src/components/pages/PrintingPage/printing.scss]: Copyright should be 2019-2020.

File: src/index.js [hunk 38]
```diff
@@ -4,6 +4,10 @@ import React from "react";
 // This serves as an entry point to the DOM and server renderers for React
 import ReactDOM from "react-dom";
 
+//> Additional libraries
+// Parallax
+import { ParallaxProvider } from 'react-scroll-parallax';
```
Comment (Aichnerc) [inline src/index.js]: Use double quotes instead of single quotes.

File: src/utilities/_variables.scss [hunk 39]
```diff
@@ -2,6 +2,8 @@ $dark: #212529;
 $red: #f61a42;
 $green: #77bd43;
 $pink: #ea4aaa;
+$agencyDark: #212529;
+$agencyRed: #f61a42;
```
Comment (Aichnerc) [inline src/utilities/_variables.scss]: Unify red and agencyRed.

**Review by Aichnerc — COMMENTED** [review]
There are alot of issues to be fixed. I wish myself happy fixing.

File: src/components/organisms/sections/Trusted/index.jsx (hunk 25 above)
Comment (pinterid) [inline src/components/organisms/sections/Trusted/index.jsx]: Missing -2020.

File: src/components/organisms/sections/Trusted/index.jsx (hunk 20 above)
Comment (Aichnerc) [reply src/components/organisms/sections/Trusted/index.jsx]: Damn. I just changed that today. 👀😂

**Review by sweggerat — APPROVED** [review]
looks bad

File: src/components/organisms/sections/Trusted/trusted.scss (hunk 26 above)
Comment (pinterid) [inline src/components/organisms/sections/Trusted/trusted.scss]: Missing -2020.

File: src/components/pages/BrandingPage/branding.scss (hunk 30 above)
Comment (pinterid) [inline src/components/pages/BrandingPage/branding.scss]: Missing -2020.

File: src/components/pages/BrandingPage/index.jsx (hunk 32 above)
Comment (pinterid) [inline src/components/pages/BrandingPage/index.jsx]: Remove new line.

File: src/components/pages/BrandingPage/index.jsx [hunk 40]
```diff
@@ -0,0 +1,367 @@
...
+        options = {
+          url: 'https://www.aichner-christian.com/public/logo_wedding.png'
+        };
+        break;
+      default:
+        break;
+    }
+
+    //> Temp output of the file URL
+    console.log(options);
```
Comment (pinterid) [inline src/components/pages/BrandingPage/index.jsx]: Is this still required?

File: src/components/pages/BrandingPage/index.jsx [hunk 41]
```diff
@@ -0,0 +1,367 @@
...
+  }
+
+  render() {
+
```
Comment (pinterid) [inline src/components/pages/BrandingPage/index.jsx]: Remove new line.

File: src/components/pages/BrandingPage/index.jsx (hunk 34 above)
Comment (pinterid) [inline src/components/pages/BrandingPage/index.jsx]: Missing -2020.

File: src/components/pages/DevelopmentPage/index.jsx (hunk 35 above)
Comment (pinterid) [inline src/components/pages/DevelopmentPage/index.jsx]: Missing -2020.

File: src/components/pages/PrintingPage/index.jsx (hunk 36 above)
Comment (pinterid) [inline src/components/pages/PrintingPage/index.jsx]: Missing -2020.

File: src/components/pages/PrintingPage/printing.scss [hunk 42]
```diff
@@ -0,0 +1,93 @@
...
+      }
+      .stepper.timeline li .step-content:before {
+        display: none;
+        border-left-width: 0;
+        border-right-width: 15px;
+        left: -15px;
+        right: auto;
+      }
+    }
+
```
Comment (pinterid) [inline src/components/pages/PrintingPage/printing.scss]: Remove new line.

File: src/components/pages/PrintingPage/printing.scss (hunk 37 above)
Comment (pinterid) [inline src/components/pages/PrintingPage/printing.scss]: Missing -2020.

**Review by pinterid — CHANGES_REQUESTED** [review]
Should be good to merge after a few changes.

**Review by LucaAllmaier — COMMENTED** [review]
Looks good, just a few comments.

File: src/App.js (hunk 1 above)
Comment (Aichnerc) [reply src/App.js:31]: This should be a general switch and not be dependent on a certain domain name.

**Review by Aichnerc — COMMENTED** [review]
Looks good.

**Review by pinterid — APPROVED** [review]
Looks good to merge.
